docs/config/README.md

```diff
@@ -2445,6 +2445,78 @@ number_threshold = 4
 symbol_threshold = 0
 ```
 
+## JJ
+
+The `jj` module reflects the current status of your Jujutsu repository. It shows
+different symbols and formats depending on the state of your work in progress,
+including changes, commits, branch, and more.
+
+The `jj` module is only shown when you're in a directory that is part of a
+Jujutsu repository.
+
+The default functionality includes displaying:
+
+- current change id
+- current commit id
+
+::: warning
+
+This module requires the Jujutsu command line tool to be installed and
+accessible in your system's PATH for it to function correctly.
+
+:::
+
+### Options
+
+| Option                   | Default                                                                                                                                                                                                                                                             | Description                                                                                  |
+| ------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------- |
+| `symbol`                 | `'🍐 '`                                                                                                                                                                                                                                                             | The string used before displaying the repository status.                                     |
+| `no_description_symbol`  | `' 📝'`                                                                                                                                                                                                                                                             | Is displayed when change description is empty.                                               |
+| `divergent_symbol`       | `' 💥'`                                                                                                                                                                                                                                                             | Is displayed when commit in divergent state.                                                 |
+| `format`                 | `'\[$symbol[$change_id_prefix]($change_id_prefix_style)[$change_id_suffix]($change_id_suffix_style) [$commit_id_prefix]($commit_id_prefix_style)[$commit_id_suffix]($commit_id_suffix_style) on [$branch]($branch_style)$no_description_symbol$divergent_symbol\]'` | The format for rendering the module. `symbol`, `style`, and other variables can be included. |
```

Comment:
The issue here is that some parts will be displayed even if there are no associated data. For example, on a change with no branch, a description and no divergence, this prompt will end with ` on `, which is not wanted

Reply:
Ah I see you worked around it with `<no branch>` but given that's the default state in JJ, I don't think adding 14 characters (`on <no-branch>`) for this case is worth it. If there is no branch, this whole part should probably no be displayed

Reply:
The issue is that you need to apply styling to both `on` and the branch name, showing `on` conditional based on the presence of the branch name.